extended/src/main/java/apoc/cypher/CypherExtended.java

```diff
@@ -82,25 +83,41 @@ public class CypherExtended {
     @Context
     public URLAccessChecker urlAccessChecker;
 
-    @Procedure(mode = WRITE)
+    @Procedure(name = "apoc.cypher.runFile", mode = WRITE)
     @Description("apoc.cypher.runFile(file or url,[{statistics:true,timeout:10,parameters:{}}]) - runs each statement in the file, all semicolon separated - currently no schema operations")
     public Stream<RowResult> runFile(@Name("file") String fileName, @Name(value = "config",defaultValue = "{}") Map<String,Object> config) {
-        return runFiles(singletonList(fileName),config);
+        return runFiles(singletonList(fileName), config);
+    }
+    
+    @Procedure(value = "apoc.cypher.runReadFile", mode = READ)
+    @Description("apoc.cypher.runReadFile(file or url,[{statistics:true,timeout:10,parameters:{}}]) - runs each `READ` statement in the file, all semicolon separated")
+    public Stream<RowResult> runReadFile(@Name("file") String fileName, @Name(value = "config",defaultValue = "{}") Map<String,Object> config) {
+        return runReadFiles(singletonList(fileName),config);
     }
 
-    @Procedure(mode = WRITE)
+    @Procedure(value = "apoc.cypher.runFiles", mode = WRITE)
     @Description("apoc.cypher.runFiles([files or urls],[{statistics:true,timeout:10,parameters:{}}])) - runs each statement in the files, all semicolon separated")
     public Stream<RowResult> runFiles(@Name("file") List<String> fileNames, @Name(value = "config",defaultValue = "{}") Map<String,Object> config) {
+        return runNonSchemaFiles(fileNames, config, true);
```

Comment:
qui abbiamo levato i parameters ma non li abbiamo levati dalla documentaazione?

Reply:
inoltre chiamando runNonSchemaFiles da runFiles non ci perdiamo le statistiche?

Reply:
I parameters vengono popolati alla riga 112.

In pratica il metodo `runNonSchemaFiles()` coincide con quello che prima stava dalla riga 94 a 97.

Ho solo aggiunto `defaultStatistics` che qui è true, mentre sulla riga 107 ha default false (quindi non stampa le statistiche, perché sono pressoché inutili con operazioni read-only).


Le 2 procedure a riga 86 e 92 semplicemente richiamano le procedure corrispettive con la s finale
- apoc.cypher.runFile --> richiama apoc.cypher.runFiles (come in precedenza)
- apoc.cypher.runReadFile --> richiama apoc.cypher.runReadFiles